### PR TITLE
Allow use of coupled chemical potential in split Cahn-Hilliard equation

### LINKED ISSUE #14140: Allow coupled use of SplitCHWRes
## Reason
To add off-diagonal transport coefficient terms we need to be able to add terms to the time derivative of a concentration c1 that use the gradient of a different chemical potential (eg w2), together with the M_12 transport coefficient in `SplitCHWResBase`.
We've had two requests on the mailing list for this and we have projects at the lab this FY that need this capability.

## Design
We'll use the same design as `SoretDiffusionBase`, where an optional `w` parameter can be passed in that replaces the use of the kernel variable when supplied.
 
## Impact
Added capability.

### PR DESCRIPTION
**Pull request by dschwen**
With this users can implement the terms containing off-diagonal the Onsager phenomenological transport matrix coefficients.

Closes #14140

### CODE REVIEW

**Comment by dschwen** [comment]
There seems to be some communication error between civet and github. All tests passed. @milljm

**Comment by milljm** [comment]
The server in which Civet is running, is slow at times. Indeed, more often than not, I have to re-send my webhook payload when I create a PR into the package_builder. This sort of communication issue seems to increase depending on how many targets there are.

**Review by permcody — DISMISSED** [review]

File: modules/phase_field/test/tests/phase_field_kernels/SplitCHWRes.i [hunk 1]
```diff
@@ -0,0 +1,133 @@
+#
+# Test the split parsed function free enery Cahn-Hilliard Bulk kernel
+# with two concentration variables and cuopling through off-diagonal Onsager
```
Comment (laagesen) [inline modules/phase_field/test/tests/phase_field_kernels/SplitCHWRes.i]: typo cuopling

File: modules/phase_field/doc/content/source/kernels/SplitCHWRes.md [hunk 2]
```diff
@@ -6,12 +6,40 @@ This kernel implements the weak form
...
+together with [`SplitCHParsed`](/SplitCHParsed.md) and
+[`CoupledTimeDerivative`](/CoupledTimeDerivative.md) to set up a system of two
+first order PDEs using a concentration order parameter and a chemical potential
+variable.
 
-For an implementation with an anisotropic (tensorial) mobility see [`SplitCHWResAniso`](/SplitCHWResAniso.md).
+## Off-diagonal Onsager Matrix
+
+The terms containing the off-diagonal components of the Onsager
+phenomenologicsal transport coefficients model the driving force resulting from
```
Comment (laagesen) [inline modules/phase_field/doc/content/source/kernels/SplitCHWRes.md]: typo phenomenologicsal

**Review by laagesen — CHANGES_REQUESTED** [review]
Looks good, just a couple of typos

**Review by permcody — APPROVED** [review]

**Pull request merged**